### ISSUE TITLE
the TPS down slowly

### ISSUE THREAD

**Issue by lingo169** [spring-projects/spring-boot#12416]
My simple object like this:
pom.xml

\<parent>
    \<groupId>org.springframework.boot\</groupId>
    \<artifactId>spring-boot-starter-parent\</artifactId>
    \<version>1.5.10.RELEASE\</version>
\</parent>
\<dependencies>
    \<dependency>
        \<groupId>org.springframework.boot\</groupId>
        \<artifactId>spring-boot-starter-web\</artifactId>
        \<exclusions>  
            \<exclusion>  
                \<groupId>org.springframework.boot\</groupId>  
                \<artifactId>spring-boot-starter-tomcat\</artifactId>  
            \</exclusion>  
        \</exclusions>
    \</dependency>
    \<dependency>
        \<groupId>org.springframework.boot\</groupId>
        \<artifactId>spring-boot-starter-undertow\</artifactId>
    \</dependency>  
\</dependencies>
\<build>
    \<plugins>
        \<plugin>
            \<groupId>org.springframework.boot\</groupId>
            \<artifactId>spring-boot-maven-plugin\</artifactId>
        \</plugin>
    \</plugins>
\</build>

**The Class like this:**

@RestController
@SpringBootApplication
public class Example {
    @RequestMapping("/")
    String home() {
        return "Hello World!__";
    }
    public static void main(String[] args) throws Exception {
        SpringApplication.run(Example.class, args);
    }
}

**Execute the command:**
`/home/gdct/jdk1.8.0_152/bin/java -server -Xms1g -Xmx1g -XX:MetaspaceSize=128M -XX:MaxMetaspaceSize=256m -jar /home/gdct/abc/aembedjar.jar --server.port=9998`

At first,the TPS is 3600 or so,But After a few minutes,THE TPS is 600 or so.(down Slowly )

Please help me,how to keep the TPS is 3500 or so ???

note: I had test a webservice api, it can keep the tps.

**This is the thread dump [github.com/lingo169/testweb/blob/master/threads9.txt](https://github.com/lingo169/testweb/blob/master/threads9.txt)**

**Comment by lingo169** [comment]
Test by soapUI,
Threads:1000
Test delay:0
Limits:6000s

**Comment by wilkinsona** [comment]
If I use SoapUI for a similar load test, it uses ~450% CPU whereas the service it's testing only reaches ~50%. As far as I can tell the performance bottleneck is SoapUI rather than the service it's testing. I suspect that this is made worse by the fact that you've configured your test to use 1000 threads which will lead to a huge amount of context switching that will only slow things down. For a true performance test, you should also run the client and server on separate machines.

In short, I can't see anything in your issue description or in my attempt to recreate the problem that suggests there's a problem with Spring Boot. If you do some further analysis and are able to identify a problem with Spring Boot then please do let us know and we'll take a look.
